components/script/dom/window.rs

```diff
@@ -2620,6 +2620,23 @@ impl Window {
         )
     }
 
+    /// Do the same kind of query as `Self::padding_top_and_left_query`, but do not force a reflow.
+    /// This is used for things like `IntersectionObserver` which should observe the value
+    /// from the most recent reflow, but do not need it to reflect the current state of
+    /// the DOM / style.
+    pub(crate) fn padding_top_and_left_query_without_reflow(
```

Comment:
If this is only used here, I guess it doesn't need to be `pub(crate)`. I'm not sure if you plan was to use this in `ResizeObsever` though...